src/core/namespace.c

```diff
@@ -3996,6 +3997,84 @@ static const char* const private_bpf_table[_PRIVATE_BPF_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(private_bpf, PrivateBPF);
 
+#include "bpf-delegate-configs.inc"
+
+DEFINE_STRING_TABLE_LOOKUP(bpf_delegate_cmd, uint64_t);
+
+int bpf_dump_delegate_commands(FILE *fd, uint64_t u) {
+        assert(fd);
+
+        if (u == UINT64_MAX)
+                fprintf(fd, "any");
+        else {
+                bool first = true;
+                for (uint64_t i = 0; i < sizeof(u) * 8; i++) {
+                        if (u & (1UL << i)) {
+                                if (!first)
+                                        fprintf(fd, ",");
+                                first = false;
+                                fprintf(fd, "%s", bpf_delegate_cmd_to_string(i));
+                        }
+                }
+        }
+
+        return 0;
+}
+
+char* bpf_delegate_commands_to_string(uint64_t u) {
+        if (u == UINT64_MAX)
+                return strdup("any");
+
+        _cleanup_free_ char *buf = NULL;
+        size_t len = 1;
+
+        for (uint64_t i = 0; i < sizeof(u) * 8; i++)
+                if (u & (1UL << i))
+                        len += strlen(bpf_delegate_cmd_table[i]) + 1;
+
+        buf = malloc(len);
```

Comment:
I do not think it is necessary to optimize here.
Please use `strextendf_with_separator()`

Reply:
Hm, but we already do like that at many places..
Rather than optimize here, I think it is better to improve `strextend()` and friends. Then we can speed up many places.
Maybe, caching the current string length should be enough, like we do in strv_extend_n().